src/ert/_c_wrappers/enkf/res_config.py

```diff
@@ -685,3 +685,22 @@ def __eq__(self, other):
 
     def __ne__(self, other):
         return not self == other
+
+    def __repr__(self):
+        return f"ResConfig({str(self)})"
+
+    def __str__(self):
+        return (
+            "{\n"
+            f"SubstConfig: {self.subst_config},\n"
+            f"SiteConfig: {self.site_config},\n"
+            f"AnalysisConfig: {self.analysis_config},\n"
+            f"ErtWorkflowList: {self.ert_workflow_list},\n"
```

Comment:
The representation still contains the `random_seed` field. Also there is a separate `repr` format for classes where the representation is not supposed to work with `eval()`, its what you get when you call e.g.
```python
>>> repr(int)
"<class 'int'>"
```
So I suggest we use:
```
    def __repr__(self):
        return (
            f"<ResConfig {self.subst_config}"
            f"{self.site_config}"
            f"{self.random_seed}"
            f"{self.analysis_config}"
            f"{self.ert_workflow_list}"
            f"{self.hook_manager}"
            f"{self.ert_templates}"
            f"{self.ecl_config}"
            f"{self.ensemble_config}"
            f"{self.model_config}"
            f"{self.queue_config}>"
        )
```
or something equivalent.